pkg/cluster/cluster.go

```diff
@@ -263,6 +265,18 @@ func (c *Cluster) run() {
 				break
 			}
 
+			// Recreate the cluster if it exceeds the scaling timeout
+			// this ensures that the operator does not get stuck in a create loop.
+			// If the etcd cluster is not accessible (e.g. due to intermittent network issues)
+			// then the operator is not able to recover from that - but this may be recoverable if pods get re-created.
+			scalingStart, err := c.status.LastTransitionTime(api.ClusterConditionScaling)
+			if c.cluster.Spec.ScalingTimeout != nil && err == nil && time.Since(scalingStart).Seconds() > c.cluster.Spec.ScalingTimeout.Seconds() {
```

Comment:
The `api.ClusterConditionScaling` condition is removed once the cluster is up - it is replaced with api.ClusterConditionAvailable.